### PR TITLE
Bp 517 implement help

### PR DESCRIPTION
**Pull request by MaximilianV**
In the header you will see an new _?_ button.
Clicking it will display several help bullets on the page explaining the functionality of single components or input fields.

### CODE REVIEW

File: src/Header.js [hunk 1]
```diff
@@ -22,6 +32,34 @@ class Header extends Component {
...
+		let hintButton = <IconHelpInactive/>;
+		let buttonTooltip = "Show help bullets on page";
+		if (this.state.hintsVisible) {
+			hintButton = <IconHelp/>;
+			buttonTooltip = "Hide help bullets on page";
+		}
+		return (
+			<IconButton
+				tooltip={buttonTooltip}
+				onTouchTap={() => this.toggleHintsOnPage()}>{hintButton}</IconButton>
```
Comment (frederike-ramin) [inline src/Header.js]: Please break this line.

File: src/Header.js [hunk 2]
```diff
@@ -22,6 +32,34 @@ class Header extends Component {
...
+		else {
+			introJs.introJs().showHints();
+			introJs.introJs().addHints();
+		}
+		this.setState({
+			hintsVisible: !hintsCurrentlyVisible
+		});
+	}
+
+	showHelpButton() {
```
Comment (frederike-ramin) [inline src/Header.js]: This is more a method to GET the help button right?

File: src/SettingsView/EntityMappingListItem.js [hunk 3]
```diff
@@ -61,13 +61,15 @@ class EntityMappingListItem extends ConnectionComponent {
...
-						<IconButton><IconEdit/></IconButton>
+						<IconButton toolTip="Edit this mapping"><IconEdit/></IconButton>
 						<ConfirmationMessage
 							actionToPerform={this.deleteMapping}
 							ref={(input) => {this.confirmationMessage = input;}}>
                             {config.messages.deleteEntityMappingMessage}
 						</ConfirmationMessage>
-						<IconButton onTouchTap={() => {this.confirmationMessage.handleOpen();}}>
+						<IconButton
+							tooltip="Delete this mapping"
```
Comment (frederike-ramin) [inline src/SettingsView/EntityMappingListItem.js:71]: If we really want to make everything configurable, this should be too.

File: src/config/config-template.js [hunk 4]
```diff
@@ -33,6 +33,19 @@ const config = {
 		deleteEntityMappingMessage: "Do you really want to delete this EntityMapping?",
 		requiredFieldMessage: 'Please fill out this field'
 	},
+	explanations: {
```
Comment (frederike-ramin) [inline src/config/config-template.js]: Why don't you put those into the help config too?

File: src/config/help-template.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+		},
+		entityMappingView: {
+			targetStatus: "If an status is selected here, each event fitting a mapping will trigger a status-change on the mapped entity.",
+			eventTypeSelection: "Select the event type where a mapping should be defined on.",
+			entityTypeSelection: "Select an entity type where the mapping should point to.",
+			attributesSelection: "Connect attributes from an event type to attributes of an entity type."
+			+ " An event will be mapped to an entity if their values match for all defined attributes.<br><br>"
+			+ "Example:<br>Given the mapping \"EventType1 > entityName to EntityType1 > name\".<br>"
+			+ "For each event of EventType1 the value of 'entityName' will be checked against each entity of EntityType1."
+			+ " If the value matches the 'name' attribute, the event will be assigned to this entity.",
```
Comment (frederike-ramin) [inline src/config/help-template.js:45]: Spaaace!

**Review by frederike-ramin — CHANGES_REQUESTED** [review]

File: src/config/help-template.js (hunk 5 above)
Comment (MaximilianV) [reply src/config/help-template.js:45]: There needs to be a space ;) Line 44 doesn't end with a line break. Only the string itself was broken down, do improve code readability. If you rebuild the string without this whitespace you would end up with `[...] EntityType1.If the [...]`.

File: src/SettingsView/EntityMappingListItem.js (hunk 3 above)
Comment (MaximilianV) [reply src/SettingsView/EntityMappingListItem.js:71]: PO said: Later.

File: src/config/config-template.js (hunk 4 above)
Comment (MaximilianV) [reply src/config/config-template.js]: Will be deleted, was some old stuff 😔

File: src/config/help-template.js (hunk 5 above)
Comment (frederike-ramin) [reply src/config/help-template.js:45]: Aaahhrg..alright.. :D

**Review by frederike-ramin — APPROVED** [review]

**Pull request merged**